### ISSUE TITLE
Fix Request: UnconditionalCreatedLogArguments should recognize record getters.

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#569]
Generated getters of records don't start with 'get', they are just the field name with added '()'.

**Comment by jborgers** [comment]
Also support record style getters like `org.apache.kafka.clients.consumer.ConsumerRecord`, assume class name ending with `Record`.

**Comment by jborgers** [comment]
Example:
```
import org.slf4j.Logger;
class Foo {
    private Logger log;

    void bar1(MyRecord obj) {
      String logObj = obj.topic();
      log.debug("record topic: {}", logObj); // <- false positive
    }

    void bar2(WannabeRecord obj) {
      String logObj = obj.topic();
      log.debug("wannabe record topic: {}", logObj); // <- false positive
      long logObj2 = obj.doStuff(5);
      log.debug("wannabe record done stuff: {}", logObj2); // bad 
    }

   void bar3() {
      WannabeRecord obj = newWR();
      String logObj = obj.topic();
      log.debug("wannabe record topic: {}", logObj); // <- false positive
    }
}

record MyRecord (
    String topic,
    Long documentId,
    long offset 
) {}

class WannabeRecord {
    String topic;
    Long documentId;
    long offset; 
    WannabeRecord(String topic, Long documentId, long offset) {
        this.topic = topic;
        this.documentId = documentId;
        this.offset = offset;
    }

    public String topic() {
        return topic;
    }
    public Long documentId() {
        return documentId;
    }
    public long offset() {
        return offset;
    }
    public long doStuff(long index) {
        long result; // = do the stuff
        return result;
    }
}
```
